jlm/hls/backend/rhls2firrtl/base-hls.hpp

```diff
@@ -72,63 +74,75 @@ class BaseHLS
   static std::string
   get_base_file_name(const llvm::RvsdgModule & rm);
 
-  std::vector<jlm::rvsdg::RegionArgument *>
-  get_mem_resps(const llvm::lambda::node * lambda)
+  /**
+   * Extracts all region arguments of the given kernel that represent memory responses.
+   * They can provide multiple values within a single execution of the region.
+   * @param lambda the lambda node holding the hls kernel
+   * @return the arguments that represent memory responses
+   */
+  std::vector<rvsdg::RegionArgument *>
+  get_mem_resps(const llvm::lambda::node & lambda)
```

Comment:
If we use the term kernel for the HLS lambda then it could be named
GetKernelMemoryResponses()

Could take the naming convention one step further:
GetKernelMemoryResponses(const llvm::lambda::node & kernel)
GetKernelMemoryResponses(const llvm::lambda::node & hlsKernel)

Reply:
We could absolutely change to something closer to the established naming scheme. I was also thinking about the fact that there is no reason for these functions to be protected methods on BaseHLS. They do not use `this` at all